app/modules/individuals/models.py

```diff
@@ -123,6 +127,55 @@ def remove_encounter(self, encounter):
     def get_members(self):
         return [encounter.owner for encounter in self.encounters]
 
+    def get_names(self):
+        return self.names
+
+    # should be only one of these
+    def get_name_for_context(self, context):
+        return Name.query.filter_by(individual_guid=self.guid, context=context).first()
+
+    # can be many
+    def get_names_for_value(self, value):
+        return Name.query.filter_by(individual_guid=self.guid, value=value).all()
+
+    def add_name(self, context, value, creator):
+        new_name = Name(
+            individual_guid=self.guid,
+            context=context,
+            value=value,
+            creator_guid=creator.guid,
+        )
+        try:
+            with db.session.begin(subtransactions=True):
+                db.session.add(new_name)
+        except (sqlalchemy.orm.exc.FlushError, sqlalchemy.exc.IntegrityError) as err:
+            log.warning(
+                f'failed to add name to {self} (context={context}, value={value}, creator={creator}: {str(err)}'
+            )
+            raise ValueError(
+                f'name could not be added, perhaps context "{context}" is already used'
+            )
+        return new_name
```

Comment:
Should this ba an audit log?

Reply:
ah yeah that makes sense.  as for the AuditLog -- not really sure it is "severe" enough to care about -- especially since i think this collision of `context` will be fairly common.    for better or worse, we are allowing only _one_ unique context across the entire individual.  so if you want the context `nickname` and get there first, you win.  but then if i come in and try to use that, it will fail here.   probably not Audit-worthy?  but i could be overridden on that call.

Reply:
cleared up in slack with english talk words.